### PR TITLE
Tcp additions

### PR DESCRIPTION
**Pull request by Qwinci**
Implement sockname/peername and FIONREAD ioctl.

### CODE REVIEW

**Review by ElectrodeYT — APPROVED** [review]
LGTM

File: servers/netserver/src/ip/tcp4.cpp [hunk 1]
```diff
@@ -214,8 +217,10 @@ struct Tcp4Socket {
 			co_return protocols::fs::Error::accessDenied;
 		}
 
-		if (!ip4().hasIp(bindEp.ipAddress))
+		if (bindEp.ipAddress != INADDR_ANY && !ip4().hasIp(bindEp.ipAddress)) {
+			std::cout << "netserver: IP address " << std::setw(8) << std::hex << bindEp.ipAddress << " is not available" << std::endl;
```
Comment (no92) [inline servers/netserver/src/ip/tcp4.cpp:221]: This should probably revert back to decimal with `std::dec`.

**Review by no92 — APPROVED** [review]
LGTM with one minor nitpick.

File: servers/netserver/src/ip/tcp4.cpp (hunk 1 above)
Comment (Qwinci) [reply servers/netserver/src/ip/tcp4.cpp]: I added that.

**Review by Dennisbonke — APPROVED** [review]
LGTM

**Pull request merged**